flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -900,7 +921,7 @@ public void heartbeatFromResourceManager(ResourceID resourceID) {
 	}
 
 	@Override
-	public CompletableFuture<TransientBlobKey> requestFileUpload(FileType fileType, Time timeout) {
+	public CompletableFuture<TransientBlobKey> requestFileUpload(FileType fileType, String fileName, Time timeout) {
```

Comment:
If `fileType` is `LOG` or `STDOUT`, it seems that the information in `fileName` won't be relevant. Is that right?

Reply:
yes.

Reply:
I think this method signature is hard to understand. Let me summarize:
- If `fileType` is `LOG` or `STDOUT`,  `fileName` will be `taskmanager.log` or `taskmanager.out`, respectively. However, `fileName` can be ignored because the file to be uploaded will be determined by `taskmanagerConfiguration`).
- If `fileType` is `CUSTOM`, the value of `fileName` should be used.

That is, the task fulfilled by of this method changes depending on the method arguments. Moreover, just by looking at the signature it is unclear what this method does. This is normally a hint to split the methods:

```
CompletableFuture<TransientBlobKey> requestFileUpload(String fileName, @RpcTimeout Time timeout);

CompletableFuture<TransientBlobKey> requestFileUpload(FileType fileType, @RpcTimeout Time timeout);
```
I think the same applies for `ResourceManagerGateway`. WDYT?

Reply:
Sounds good, I'll update it.

Reply:
Ok, the new method names make sense.